### DISCUSSION TITLE
Catch errors in *Parse* from serial connection

### DISCUSSION THREAD

**Discussion by horsto** [bonsai-rx/bonsai#954] [Q&A] [answered]
I am reading incoming data on a serial port on Windows (coming from a USB connected microcontroller). I then parse the data and extract items that I then write to a .csv file. 
In very rare cases the microcontroller sends data that is "just" not fitting the specified format in _Parse_. This then throws an exception and crashes the whole workflow. 
Is there a way to catch the error / skip it so the whole thing continues working? 

<img width="291" alt="serial" src="https://user-images.githubusercontent.com/7548928/176963114-66e19333-82c3-40f8-acff-59f00aee8e48.PNG">

**Comment by bruno-f-cruz** [comment, marked as answer]
Depending on how you want to handle the errors you could try to use the `Retry`, `RetryCount`, `Catch`, or even a custom `PythonTransform` operator.

**Reply by horsto** [reply]
Great, thanks - do you have an example workflow for *Catch* (at least from the description it seems like the thing I want, but it cannot create a connection forward?)

**Reply by bruno-f-cruz** [reply]
In your case I think `Retry` might be the best. Simply:
![image](https://user-images.githubusercontent.com/7049351/177013448-e712671e-b55c-4e8f-ba55-333e1bdf8b55.png)

Retry wont close the branch and instead wait for a valid output.

**Reply by horsto** [reply]
Perfect, thank you!

**Reply by yqwang1** [reply]
same question here, Retry worked well, thanks!!!
